### PR TITLE
Crate Shelves

### PR DESCRIPTION
**Pull request by west3436**
## What this does

Ports Crate Shelves from https://github.com/shiptest-ss13/Shiptest/pull/2374. Using modified sprites with original sprite credit to @Imaginos16.

image

demo

crash

Shelves can be crafted with 3 metal sheets and are assembled and deconstructed the same way racks are. Three crates can be stacked on a shelf and cannot be opened until they are pulled off the shelf. Deconstructing an assembled shelf will drop the crates to the ground with a 5% chance of breaking one of the crates open. An admin-only extended version can stack up to 12 crates high for a very silly visual effect. Bumping into the shelves, hitting them with vehicles, or throwing things at them pose a risk of collapse - especially if some dirty dog unscrewed the bolts beforehand!

## Why it's good

Gives QMs and Cargo Techs more power to organize their junk.

## How it was tested

- Crafting shelves.
- Assembling and disassembling.
- Dragging crates on and off.
- Deconstructing the shelf with crates on it.

## Changelog

:cl:
 * rscadd: Added Crate Shelves which can be used to stack multiple crates in a single tile.

### CODE REVIEW

**Comment by GenericDM** [comment]
Hello! Thank you for porting crate shelves, it's always a great feeling to see that someone thought my code was worth bringing somewhere else. Just so you know, the sprites were actually made by PositiveEntropy aka @imaginos16
Also, if you haven't already, I'd suggest taking a look at the commit history of the file over on shiptest so that you can make sure you have the patches I made to fix some bugs.
Hope you enjoy the shelves!

**Comment by west3436** [comment]
> Hello! Thank you for porting crate shelves, it's always a great feeling to see that someone thought my code was worth bringing somewhere else. Just so you know, the sprites were actually made by PositiveEntropy aka @Imaginos16 Also, if you haven't already, I'd suggest taking a look at the commit history of the file over on shiptest so that you can make sure you have the patches I made to fix some bugs. Hope you enjoy the shelves!

Thanks - credit updated!

**Comment by falcon2346** [comment]
I feel like this doesn't have a high enough chance for horrific industrial accidents. I'd love to see a chance of it falling over and crushing some poor cargo tech if it's kicked

**Comment by ErrorsWindows** [comment]
^ give the ability to unscrew (not unwrench) some parts and have it squish somebody like a vending machine

**Comment by SonixApache** [comment]
map one into each cargo warehouse to replace that fucking useless rack with
wonder how hard it would be to add mapped variants with pre-inserted crates... probably not hard.

File: code/game/objects/items/stacks/stack_recipes.dm [hunk 1]
```diff
@@ -340,6 +340,7 @@ var/list/datum/stack_recipe/metal_recipes = list (
 		), 2),
 	new/datum/stack_recipe("table parts", /obj/item/weapon/table_parts,                           2                                ),
 	new/datum/stack_recipe("rack parts",  /obj/item/weapon/rack_parts,                                                             ),
+	new/datum/stack_recipe("crate shelf parts", /obj/item/weapon/rack_parts/shelf,                3                                ),
```
Comment (jwhitak) [inline code/game/objects/items/stacks/stack_recipes.dm]: Looking at the sprite, clearly, this rack needs to cost 4 rods in addition to the 3 metal sheet requirement.........................

Comment (west3436) [reply code/game/objects/items/stacks/stack_recipes.dm]: Increased to 5 sheets

**Pull request merged**